### PR TITLE
Remove Decimal and replace strason with serde_json

### PR DESCRIPTION
**Pull request by stevenroose**
I'm also wondering if it makes sense to have Decimal in here at all.. Was that intended to be used for difficulty numbers or so? In rust-bitcoincore-rpc we're using `BigUint` from the [num-bigint](https://crates.io/crates/num-bigint) crate for that.

### CODE REVIEW

File: src/util/decimal.rs [hunk 1]
```diff
@@ -175,54 +175,47 @@ impl Decimal {
...
     type Err = ParseDecimalError;
 
     /// Parses a `Decimal` from the given amount string.
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        Json::from_str(s)?
-            .num()
-            .ok_or(ParseDecimalError::NotANumber)
-            .and_then(Decimal::parse_decimal)
+        Decimal::parse_decimal(&serde_json::Number::from_str(s)?.to_string())
+        //serde_json::Number::from_str(s)
```
Comment (sgeisler) [inline src/util/decimal.rs]: I guess you just forgot that in here?

**Comment by stevenroose** [comment]
Yeah I forgot that. It is compiling for me, I considered it quite finished.

Just somehow Travis doesn't seem to pull in dev-dependencies. I added
serde_json to dev-dependencies (and as an optional regular dependency), so
the test run should have it. I'll look into it later.

File: src/util/decimal.rs (hunk 1 above)
Comment (stevenroose) [reply src/util/decimal.rs]: fixed

File: src/util/psbt/mod.rs [hunk 2]
```diff
@@ -116,7 +116,7 @@ impl<D: Decoder> Decodable<D> for PartiallySignedTransaction {
             return Err(Error::InvalidMagic.into());
         }
 
-        if 0xff_u8 != Decodable::consensus_decode(d)? {
+        if 0xff != u8::consensus_decode(d)? {
```
Comment (stevenroose) [inline src/util/psbt/mod.rs]: Just wanted to mention that this somehow broke for me. It couldn't infer type, even though the 0xff_u8 is very explicitly typed. Same goes for the misc.rs change. Well, never hurts to be more specific I guess.

**Comment by stevenroose** [comment]
Would be nice to have this merged together with https://github.com/rust-bitcoin/rust-bitcoin/pull/252.

**Comment by dpc** [comment]
Maybe I don't understand something and it's not an issue, but just thinking aloud:

I don't understand how the "arbitrary_precision" feature on `serde_json`, but it looks too me like a potential misfeature. Since features in `cargo` are accumulative, once someone includes this library in their project, it will enable `arbitrary_precision` on `serde_json` for **all libraries**. What if some piece of code (maybe accidently) depending on the loss of precision?

**Comment by stevenroose** [comment]
@dpc `arbitrary_precision` is only enabled when you're using `Decimal`, which I think can be removed once `Amount` [lands](https://github.com/rust-bitcoin/rust-bitcoin/pull/252). 

@apoelstra any other objection to this?

**Comment by apoelstra** [comment]
Cool. Let's get `Amount` in so we can remove `arbitrary_precision` and then I'll take a look at this.

Frustrating that we can't feature-gate `dev-dependencies`, it would eliminate any concern about downstream users being affected by our feature choices.

**Comment by stevenroose** [comment]
So you prefer to have Amount come first? Because currently it builds on this PR. I could change that though. 
Thing is that this PR also does some [dev-]dependency cleanup that I'd want to do for Amount as well.

**Comment by apoelstra** [comment]
Can this be rebased now that #270 is in?

**Comment by stevenroose** [comment]
I also removed Decimal, is that ok?

File: src/blockdata/script.rs [hunk 3]
```diff
@@ -805,15 +805,14 @@ mod test {
     }
 
     #[test]
-    #[cfg(all(feature = "serde", feature = "strason"))]
+    #[cfg(feature = "serde")]
```
Comment (apoelstra) [inline src/blockdata/script.rs]: Needs to be `feature = "serde_json"`, no?

Comment (stevenroose) [reply src/blockdata/script.rs]: No, you're testing the serde stuff in the main code and serde_json is a dev-dependency that is always present for testing.

(I was mistaken in the Cargo file, though, I forgot to move serde_json to dev-dependencies. We don't rely on it for anything else than tests.)

**Comment by apoelstra** [comment]
No coordination, we'll put it all in the next major release.

**Pull request merged**